### PR TITLE
Feat(optimizer): improve struct type annotation support for EQ-delimited kv pairs

### PR DESCRIPTION
**Pull request by fool1280**
Handle `SELECT STRUCT_PACK(a := 1, b := 2.5)`

### CODE REVIEW

File: tests/test_optimizer.py [hunk 1]
```diff
@@ -550,18 +550,26 @@ def test_scope_warning(self, logger):
...
+                "SELECT STRUCT(1 AS col, 2.5, ARRAY[1, 2, 3] AS nested_array, 'foo')",
+                "bigquery",
+            ): "STRUCT<col INT, DOUBLE, nested_array ARRAY<INT>, VARCHAR>",
+            ("SELECT STRUCT(1, 2.5, 'bar')", "spark"): "STRUCT<INT, DOUBLE, VARCHAR>",
+            ('SELECT STRUCT(1 AS "CaseSensitive")', "spark"): 'STRUCT<"CaseSensitive" INT>',
+            ("SELECT STRUCT_PACK(a := 1, b := 2.5)", "duckdb"): "STRUCT<a INT, b DOUBLE>",
+            ("SELECT ROW(1, 2.5, 'foo')", "presto"): "STRUCT<INT, DOUBLE, VARCHAR>",
         }
 
         for sql, target_type in tests.items():
```
Comment (georgesittas) [inline tests/test_optimizer.py:570]:
```suggestion
        for (sql, dialect), target_type in tests.items():
```

File: tests/test_optimizer.py [hunk 2]
```diff
@@ -550,18 +550,26 @@ def test_scope_warning(self, logger):
...
             with self.subTest(sql):
-                expression = annotate_types(parse_one(sql))
+                expression = annotate_types(parse_one(sql[0], sql[1]))
```
Comment (georgesittas) [inline tests/test_optimizer.py:572]:
```suggestion
                expression = annotate_types(parse_one(sql, read=dialect))
```

File: sqlglot/optimizer/annotate_types.py [hunk 3]
```diff
@@ -479,6 +479,18 @@ def _annotate_with_type(self, expression: E, target_type: exp.DataType.Type) ->
         self._set_type(expression, target_type)
         return self._annotate_args(expression)
 
+    @t.no_type_check
+    def _annotate_struct_value(self, expression: E) -> t.Optional[exp.DataType] | exp.ColumnDef:
```
Comment (georgesittas) [inline sqlglot/optimizer/annotate_types.py:483]:
```suggestion
    def _annotate_struct_value(
        self, expression: exp.Expression
    ) -> t.Optional[exp.DataType] | exp.ColumnDef:
```

**Review by georgesittas — APPROVED** [review]

**Review by tobymao — APPROVED** [review]

**Comment by georgesittas** [comment]
@fool1280 take a look at https://www.conventionalcommits.org/en/v1.0.0/ to properly format future PR / commit titles

**Comment by fool1280** [comment]
thank you

**Pull request merged**